data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -429,6 +429,45 @@ class ApiServerClient {
       });
   }
 
+  /**
+   * @description Get Resource in Apiserver with the opts
+   * @param {string} opts.resourceGroup - Unique id of resource
+   * @param {string} opts.resourceType - Name of operation
+   * @param {string} opts.stateList - State of resorce
+   */
+  getResourceListByState(opts) {
+    logger.debug('Get resource with opts: ', opts);
+    assert.ok(opts.resourceGroup, `Property 'resourceGroup' is required to get resource list`);
+    assert.ok(opts.resourceType, `Property 'resourceType' is required to get resource list`);
+    assert.ok(opts.stateList, `Property 'stateList' is required to fetch resource list`);
+    return Promise.try(() => this.init())
+      .then(() => apiserver.apis[opts.resourceGroup][CONST.APISERVER.API_VERSION]
+        .namespaces(CONST.APISERVER.NAMESPACE)[opts.resourceType].get({
+          qs: {
+            labelSelector: `state in (${_.join(opts.stateList, ',')})`
+          }
+        }))
+      .then(resources => _.get(resources.body, 'items'))
+      .map(resource => {
+        _.forEach(resource.spec, (val, key) => {
```

Comment:
Would it be possible for you to iterate asynchronously here to avoid blocking the event loop?

Reply:
@abh1kg `_.forEach(resource.spec, (val, key) => {` this part is just about array looping of few objects, is there any I/O call involved internally ?
[this](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/data-access-layer/eventmesh/ApiServerClient.js#L411) is also similar.

Reply:
Even though it does not involve I/O, it still blocks the event loop since the iterative code runs synchronously. Now, we might say that the number of iterations would be small enough to make a significant difference to event loop performance but theoretically, this is still blocking. I'm fine with letting it remain as is  but an example for converting a synchronous for loop to a non-blocking one is better to be looked at. This is where async/await would help if we are willing to refactor